kamelet-support/src/main/java/io/kaoto/backend/KamelPopulator.java

```diff
@@ -201,21 +228,21 @@ private void setSpecDependencies(final KameletSpec spec,
         }
     }
 
-    private void setParameters(final List<Parameter> parameters,
-                               final KameletDefinition def) {
+    private void setParameters(final List<Parameter> parameters, final Definition def) {
         for (Parameter p : parameters) {
             //this will override anything that comes from the metadata set
             //which means there are edited changes
-            KameletDefinitionProperty property =
-                    new KameletDefinitionProperty();
+            Properties property = new Properties();
             if (p.getDefaultValue() != null) {
-                property.setDefault(p.getDefaultValue().toString());
+                AnyType defaultValue = new AnyType();
+                defaultValue.setValue(p.getDefaultValue());
+                property.set_default(defaultValue);
             }
             property.setDescription(p.getDescription());
             if ("string".equalsIgnoreCase(p.getType())) {
                 property.setFormat(((StringParameter) p).getFormat());
             }
-            property.setPath(false);
+            // property.setPath(false);
```

Comment:
commented out code?

Reply:
Maybe I can just remove it, now that I think of it, the git history is always there.